netjson-monitoring.lua

```diff
@@ -72,6 +72,31 @@ netjson = {
     }
 }
 
+-- collect interface addresses
+function get_addresses(name)
+    interface_data = ubus:call('network.interface', 'dump', {})
+    addresses = {}
+    interface_list = interface_data['interface']
+    for _, interface in pairs(interface_list) do
+        if interface['l3_device'] == name then
+            if interface['ipv4-address'] then
+                for _,address in pairs(interface['ipv4-address']) do
+                    table.insert(addresses, address)
+                end
+            elseif interface['ipv6-address'] then
+                for _,address in pairs(interface['ipv6-address']) do
+                    table.insert(addresses, address)
+                end
+            end
+            if interface.proto == 'static' then
+                addresses[#addresses].proto = 'static'
+            else
+                addresses[#addresses].proto = 'dhcp'
```

Comment:
At the moment `proto` looks like it's only added for the last element in the address array.

I propose to use the following:

```
        if interface['l3_device'] == name then
            if interface['ipv4-address'] then
                for _,address in pairs(interface['ipv4-address']) do
                    if address['proto'] == 'static' then
                        proto = 'static'
                    else
                        -- expected values are: dhcp, dhcpv6
                        proto = 'dhcp'
                    table.insert(addresses, {
                        address = address['address'],
                        mask = address['mask'],
                        proto = proto,
                        family = 'ipv4'
                    })
                end
            elseif interface['ipv6-address'] then
                .. similar for ipv6
```